File: benchmarks/cross-lang/cpu-string-build/go/bench.go
// Phase 31: cpu-string-build (Go)

package main

import (
	"fmt"
	"os"
	"strings"
)

func main() {
	const iters = 500
	total := 0
	for k := 0; k < iters; k++ {
		var b strings.Builder
		b.Grow(200)
		for i := 0; i < 100; i++ {
			b.WriteString("x|")
		}
		total += b.Len()
	}
	if total != 100000 {
		fmt.Fprintf(os.Stderr, "unexpected: %d\n", total)
		os.Exit(1)
	}
}

File: benchmarks/cross-lang/cpu-string-build/java/Bench.java
// Phase 31: cpu-string-build (Java)

public class Bench {
    public static void main(String[] args) {
        final int iters = 500;
        long total = 0L;
        for (int k = 0; k < iters; k++) {
            StringBuilder b = new StringBuilder(200);
            for (int i = 0; i < 100; i++) {
                b.append("x|");
            }
            total += b.length();
        }
        if (total != 100000L) {
            System.err.println("unexpected: " + total);
            System.exit(1);
        }
    }
}
